async_service/trio.py

```diff
@@ -263,7 +269,11 @@ def cancel(self) -> None:
                             task,
                         )
                         self.cancel()
-                        raise DaemonTaskExit(f"Daemon task {name} exited")
+                        try:
+                            raise DaemonTaskExit(f"Daemon task {name} exited")
+                        except DaemonTaskExit:
+                            self._errors.append(cast(EXC_INFO, sys.exc_info()))
+                            return
```

Comment:
This is a really odd mechanic but if `DaemonTaskExit` is raised here, then the `cancel_scope` ends up getting cancelled and that immediately causes **all** things under this task to be immediately cancelled which isn't the desired behavior.

Reply:
is it possible (and then is it clearer?) to use shielding on the cancellation scope?